src/main/kotlin/lotto/domin/WinningStatics.kt

```diff
@@ -0,0 +1,52 @@
+package lotto.domin
+
+import lotto.dto.InputPaymentRequestDto
+import lotto.dto.InputWinningLottoNumberDto
+import lotto.dto.WinningStaticsResponseDto
+import kotlin.math.floor
+
+class WinningStatics(
+    paymentDto: InputPaymentRequestDto,
+    lastWeekWinningLottoDto: InputWinningLottoNumberDto,
+    private val lottoRecord: List<LottoNumberSet>
+) {
+    private val payment: Int = paymentDto.payment
+    private val winningLottoNumber: List<Int> = lastWeekWinningLottoDto.lasWeekWinningNumber
+
+    fun run(): WinningStaticsResponseDto {
+        val containCountList: List<Int> = lottoRecord.map {
+            containsCount(it)
+        }
+
+        val matchWinningLotto: Map<LottoWinningAmount, Int> = LottoWinningAmount.values()
+            .associateWith { lottoWinningAmount ->
+                containCountList.count { it == lottoWinningAmount.matchCount }
+            }
+
+        val profitRatio: Double = profitRatio1(matchWinningLotto)
+
+        return WinningStaticsResponseDto(matchWinningLotto, profitRatio)
+    }
+
+    private fun containsCount(lottoNumberSet: LottoNumberSet): Int = lottoNumberSet.lotto
+        .sumOf {
+            if (winningLottoNumber.contains(it)) 1.toInt() else 0
```

Comment:
```suggestion
            if (winningLottoNumber.contains(it)) 1 else 0
```
toInt는 제거해도 될 것 같아요.

Reply:
toInt를 넣은 이유는 코틀린 자체에서 sumOf 함수 자체적인 문제로 double과 int를 추론하지 못해 toInt를 적어주었습니다. !!
코틀린 해당 이슈 관련되어 링크 첨부드립니다.
- https://youtrack.jetbrains.com/issue/KT-46360 
위 링크에서 Type: Problem으로 문제가 해결되지 않은 것을 확인했습니다!
- https://discuss.kotlinlang.org/t/sumof-resolution-ambiguity/22311

Reply:
제 IDE 환경에서는 toInt를 제거해도 잘 동작되서 코맨트를 달았네요 😭